### PR TITLE
Zipformer recipe for SPGISpeech

### PR DESCRIPTION
**Pull request by marcoyang1998**
This PR adds a Zipformer recipe for SPGISpeech. The model achieves the following WERs on the test sets:

|Epoch-avg | dev | val | 
| ----- | ----| ----- | 
| 30-10 | 2.08 | 2.14 |

### CODE REVIEW

File: egs/spgispeech/ASR/zipformer/model.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
+# Copyright    2021-2023  Xiaomi Corp.        (authors: Fangjun Kuang,
```
Comment (pkufool) [inline egs/spgispeech/ASR/zipformer/model.py:1]: Could we symbol link this file to existing one?

File: egs/spgispeech/ASR/RESULTS.md [hunk 2]
```diff
@@ -1,5 +1,71 @@
...
+
+Transducer: Zipformer encoder + stateless decoder.
+
+The WERs are:
+
+|                           | dev | val | comment                                  |
+|---------------------------|------------|------------|------------------------------------------|
+| greedy search             | 2.08       | 2.14       | --epoch 30 --avg 10 |
+| modified beam search      | 2.05       | 2.09       | --epoch 30 --avg 10 --beam-size 4 |
+n| fast beam search          | 2.07       | 2.17       | --epoch 30 --avg 10 --beam 20 --max-contexts 8 --max-states 64 |
```
Comment (pkufool) [inline egs/spgispeech/ASR/RESULTS.md:17]: Delete the prefix `n`.

File: egs/spgispeech/ASR/RESULTS.md [hunk 3]
```diff
@@ -1,5 +1,71 @@
...
+
+The decoding command is:
+```
+# greedy search
+python ./zipformer/decode.py \
+            --epoch $epoch \
+            --avg $avg \
+            --exp-dir ./zipformer/exp \
+            --max-duration 1000 \
+            --decoding-method modified_beam_search
```
Comment (pkufool) [inline egs/spgispeech/ASR/RESULTS.md:46]: delete this line.

**Review by pkufool — APPROVED** [review]
LGTM

File: egs/spgispeech/ASR/zipformer/model.py (hunk 1 above)
Comment (marcoyang1998) [reply egs/spgispeech/ASR/zipformer/model.py:1]: Yes, will change it.

**Comment by marcoyang1998** [comment]
Will upload the pre-trained model and onnx models in the next PR.

**Pull request merged**